### PR TITLE
feat:Migrated Batch 1 of Vapour Thermal Conductivity

### PR DESCRIPTION
**Pull request by qur786**

### CODE REVIEW

File: src/data/vapour-thermal-conductivity.ts [hunk 1]
```diff
@@ -0,0 +1,778 @@
...
+    typeof vapourThermalConductivityProps.minimumTemperature === "number" &&
+    typeof vapourThermalConductivityProps.maximumTemperature === "number" &&
+    temperature >= vapourThermalConductivityProps.minimumTemperature &&
+    temperature <= vapourThermalConductivityProps.maximumTemperature
+  ) {
+    thermalConductivity =
+      (vapourThermalConductivityProps.C1 * Math.pow(temperature, vapourThermalConductivityProps.C2)) /
+      (1 +
+        (Number.isFinite(vapourThermalConductivityProps.C3)
+          ? (vapourThermalConductivityProps.C3 as number) / temperature
```
Comment (saikat-samanta) [inline src/data/vapour-thermal-conductivity.ts:744]: Avoid type assertion
suggested: `vapourThermalConductivityProps.C3 ? 0`
so on...

**Review by saikat-samanta — COMMENTED** [review]
Rest good 👍

File: src/data/vapor-thermal-conductivity-constants.ts [hunk 2]
```diff
@@ -0,0 +1,778 @@
...
+  }
+  return thermalConductivity;
+}
+/* function to calculate thermal coductivity of acetic acid, butyric acid, formic acid, heptanoic acid, octanoic acid, pentanoic acid, propionic acid,
+
+export function calculateDIPPRThermalConductivityException(compound: string, temperature: number): number {
+  const vaporThermalConductivityProps: DIPPRVaporThermalConductivityProps | undefined =
+    DIPPRVaporThermalConductivityDictionary[compound];
+  let thermalConductivity = NaN;
+  if (
```
Comment (maneetgoyal) [inline src/data/vapor-thermal-conductivity-constants.ts:758]: - [ ] Uncomment this function
- [ ] Call this function inside `calculateDIPPRVaporThermalConductivity`, such that it gets fired only for those "special" compounds.
- [ ] Uncomment all the special compounds

**Comment by qur786** [comment]
some issues :
1. AceticAcidExtended2 tests are failing for both type of formula
2. ButyricAcidExtended test were failing for special function but for the general function , it is passing . So I have excluded  ButyricAcidExtended from speacial compounds.

File: src/data/vapor-thermal-conductivity-constants.ts [hunk 3]
```diff
@@ -0,0 +1,766 @@
...
+    thermalConductivityAtMinimumTemperature: 0.00887,
+    thermalConductivityAtMaximumTemperature: 0.09025
+  }
+};
+
+export function calculateDIPPRVaporThermalConductivity(compound: string, temperature: number): number {
+  const vaporThermalConductivityProps: DIPPRVaporThermalConductivityProps | undefined =
+    DIPPRVaporThermalConductivityDictionary[compound];
+  let thermalConductivity = NaN;
+  let compoundExceptionList = ["AceticAcid","AceticAcidExtended1", "ButyricAcid", "FormicAcid", "HeptanoicAcid", "OctanoicAcid", "PentanoicAcid", "PropionicAcid"];
```
Comment (maneetgoyal) [inline src/data/vapor-thermal-conductivity-constants.ts:724]: - Why is `let` used?
- Please run linter here. Seems like there's some formatting issue.

File: src/data/vapor-thermal-conductivity-constants.ts [hunk 4]
```diff
@@ -0,0 +1,766 @@
...
+      thermalConductivity =
+        vaporThermalConductivityProps.C1 +
+        vaporThermalConductivityProps.C2 * temperature +
+        (Number.isFinite(vaporThermalConductivityProps.C3)
+          ? Math.pow(temperature, 2) * (vaporThermalConductivityProps.C3 as number)
+          : 0) +
+        (Number.isFinite(vaporThermalConductivityProps.C4)
+          ? Math.pow(temperature, 3) * (vaporThermalConductivityProps.C4 as number)
+          : 0)
+    }
```
Comment (maneetgoyal) [inline src/data/vapor-thermal-conductivity-constants.ts:742]: Segregate it into a new small function for modularity and then call that function inside this function.

File: src/data/vapor-thermal-conductivity-constants.ts [hunk 5]
```diff
@@ -0,0 +1,766 @@
...
+      typeof vaporThermalConductivityProps.maximumTemperature === "number" &&
+      temperature >= vaporThermalConductivityProps.minimumTemperature &&
+      temperature <= vaporThermalConductivityProps.maximumTemperature
+    ) {
+      thermalConductivity =
+        (vaporThermalConductivityProps.C1 * Math.pow(temperature, vaporThermalConductivityProps.C2)) /
+        (1 +
+          (Number.isFinite(vaporThermalConductivityProps.C3)
+            ? (vaporThermalConductivityProps.C3 as number) / temperature
+            : 0) +
```
Comment (maneetgoyal) [inline src/data/vapor-thermal-conductivity-constants.ts:757]: Segregate it into a new small function for modularity.

File: src/data/vapor-thermal-conductivity-constants.ts (hunk 4 above)
Comment (maneetgoyal) [reply src/data/vapor-thermal-conductivity-constants.ts:742]: You can refer to `density-constants.ts` if need more clarity.

**Comment by qur786** [comment]
@maneetgoyal  sir , test cases are failing for AceticAcidExtended2 that's why I commented it out.

**Comment by maneetgoyal** [comment]
> @maneetgoyal sir , test cases are failing for AceticAcidExtended2 that's why I commented it out.

That's fine, try to refactor the code as suggested above for now.

**Comment by maneetgoyal** [comment]
@qur786 Have made some changes. Please study them and add more test cases.

File: test/calculator.test.ts [hunk 6]
```diff
@@ -214,9 +239,39 @@ describe("calculateDIPPRVaporThermalConductivity", () => {
...
+    );
     expect(calculateDIPPRVaporThermalConductivity("ButyricAcid", vaporPropsExtended1.maximumTemperature)).toBeCloseTo(
       vaporPropsExtended1.thermalConductivityAtMaximumTemperature,
       3
     );
   });
+
+  it("should return NaN outside the specified range of temperature", () => {
+    for (let compound in DIPPRVaporThermalConductivityDictionary) {
+      if (!specialCompoundList.includes(compound)) {
```
Comment (maneetgoyal) [inline test/calculator.test.ts:254]: `specialCompoundList.includes(compound) === false`. Prefer explicit checks.

File: test/calculator.test.ts [hunk 7]
```diff
@@ -214,9 +239,39 @@ describe("calculateDIPPRVaporThermalConductivity", () => {
...
+        const vaporProps = DIPPRVaporThermalConductivityDictionary[compound];
+        expect(calculateDIPPRVaporThermalConductivity(compound, vaporProps.minimumTemperature - 1)).toBe(NaN);
+        expect(calculateDIPPRVaporThermalConductivity(compound, vaporProps.maximumTemperature + 1)).toBe(NaN);
+      }
+    }
+  });
+
+  it("should return correct thermal conductivity at Tmin and Tmax", () => {
+    for (let compound in DIPPRVaporThermalConductivityDictionary) {
+      if (!specialCompoundList.includes(compound)) {
```
Comment (maneetgoyal) [inline test/calculator.test.ts:264]: https://github.com/manufac-analytics/perry/pull/69/commits/4a905eb4aa2651dcb4568f8153b83bd71bbce89e#r690219779

**Comment by maneetgoyal** [comment]
@qur786 Anything else pending here? (Apart from the 2 new comments)

**Comment by qur786** [comment]
> @qur786 Anything else pending here? (Apart from the 2 new comments)

No sir have you looked at NaN test for acetic acid and butyric acid

**Comment by maneetgoyal** [comment]
>have you looked at NaN test for acetic acid and butyric acid

@qur786 They are looking good. Lets address the pending comments and merge this

**Comment by saikat-samanta** [comment]
@qur786, tests look okay. Passing in my local system.
Screenshot_4

**Review by saikat-samanta — APPROVED** [review]
👍

**Review by maneetgoyal — APPROVED** [review]

**Pull request merged**